cpp/autosar/src/rules/A16-0-1/PreProcessorShallOnlyBeUsedForCertainDirectivesPatterns.ql

```diff
@@ -40,10 +53,13 @@ pragma[noinline]
 predicate isPreprocConditionalRange(
   PreprocessorBranch pb, string filepath, int startLine, int endLine
 ) {
-  exists(PreprocessorEndif end | pb.getEndIf() = end |
-    isPreprocFileAndLine(pb, filepath, startLine) and
-    isPreprocFileAndLine(end, filepath, endLine)
-  )
+  isPreprocFileAndLine(pb, filepath, startLine) and
+  endLine =
+    min(int elifEndLine, PreprocessorDirective end |
+      pb.getNext() = end and isPreprocFileAndLine(end, filepath, elifEndLine)
```

Comment:
I wasn't aware of `getNext`. This seems we don't need the `min` construct in that case.

Reply:
agree, nice catch